### PR TITLE
Cetus & Earth times, News Panel

### PR DESCRIPTION
**Pull request by Bitti09**
### Warframe Hub Pull Request
---
**Summary (short):**

- Adding Earth/Cetus Timer Panel, & switching to v-bind when usefull
- Adding NewsBox & Simple Grid Layout
---
**Description:**

---
**Fixes issue (include link):**

News component: https://github.com/WFCD/warframe-hub/issues/187

---
**Mockups, screenshots, evidence:**

---

(Check one)
- [ ] Issue fix
- [ ] Suggestion fulfillment

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -7,104 +7,89 @@
 </template>
 
 <style>
-  #app {
```
Comment (TobiTenno) [inline src/App.vue]: please revert this

File: src/App.vue [hunk 2]
```diff
@@ -7,104 +7,89 @@
...
+}
+.row {
+  display: inline;
+}
 </style>
 
 <script>
-  import Navbar from '@/components/Navbar.vue'
-  import Footer from '@/components/Footer.vue'
+import Navbar from '@/components/Navbar.vue'
```
Comment (TobiTenno) [inline src/App.vue]: keep the existing formatting

File: src/components/FishImg.vue [hunk 3]
```diff
@@ -9,23 +9,23 @@ export default {
   props: {
     type: {
       type: String,
-      required: true,
+      required: true
     },
     item: {
       type: String,
-      required: true,
+      required: true
```
Comment (TobiTenno) [inline src/components/FishImg.vue]: leave trailing commas

File: src/components/FishImg.vue [hunk 4]
```diff
@@ -87,14 +87,13 @@ export default {
     }
   },
   mounted: function() {
-    this.getImg();
```
Comment (TobiTenno) [inline src/components/FishImg.vue]: i don't know why so much formatting was changed. please revert them.

File: src/App.vue (hunk 1 above)
Comment (dsinn) [inline src/App.vue]: Please change the indentation back.

File: src/components/Navbar.vue [hunk 5]
```diff
@@ -81,60 +81,60 @@
...
 <style scoped>
 .navbar.navbar-dark.bg-primary {
-  background-color: #1a5072!important;
+  background-color: #1a5072 !important;
 }
 </style>
 <!-- Navbar.vue -->
 
 <script>
-  import HubImg from '@/components/HubImg.vue'
```
Comment (TobiTenno) [inline src/components/Navbar.vue]: as above, please maintain style that was present

File: src/components/TimeBadge.vue [hunk 6]
```diff
@@ -16,103 +16,109 @@ export default {
   props: {
     starttime: {
       type: String,
-      default: '1970-01-01T00:00:00.000Z',
```
Comment (TobiTenno) [inline src/components/TimeBadge.vue]: all of the changes in this file were unnecessary, and if there were any, they were lost in the comma removals

File: src/components/TimeBadge.vue [hunk 7]
```diff
@@ -16,103 +16,109 @@ export default {
...
-      let timeLeft = diff;
-      const stringArray = [];
+      let timeLeft = diff
+      const stringArray = []
 
-      [[86400000, 'd'], [3600000, 'h'], [60000, 'm'], [1000, 's']]
-        .forEach(([unit, suffix]) => {
-          const time = Math.floor(timeLeft / unit);
-          const first = stringArray.length === 0;
+      ;[[86400000, 'd'], [3600000, 'h'], [60000, 'm'], [1000, 's']].forEach(
```
Comment (dsinn) [inline src/components/TimeBadge.vue]: Semicolon at the beginning of a line? Some weird style changes are happening here.

File: src/components/TimeBadge.vue [hunk 8]
```diff
@@ -16,103 +16,109 @@ export default {
...
+        this.disp = `Expired: ${this.formatDurationShort(duration)}`
       } else {
-        if (diff < 600000) { // 0 min to 10 min
-          this.mutableVariant = 'danger';
-        } else if (diff < 1800000) { // 10 min to 30 min
-          this.mutableVariant = 'warning';
-        } else if (diff > 1800000) { // 30 min to 1 hour
-          this.mutableVariant = 'success';
+        if (diff < 600000) {
+          // 0 min to 10 min
```
Comment (dsinn) [inline src/components/TimeBadge.vue]: These are more code style changes that seem unnecessary to me.

File: src/components/panels/TimePanel.vue [hunk 9]
```diff
@@ -0,0 +1,72 @@
+<template>
+  <b-col md="6">
+            <h3 class="text-center">{{headertext}}</h3>
```
Comment (TobiTenno) [inline src/components/panels/TimePanel.vue]: tabs? spaces? looks like possible mismatch

File: src/main.js [hunk 10]
```diff
@@ -25,4 +25,6 @@ new Vue({
   render: h => h(App)
 }).$mount('#app')
 
-setInterval(() => {store.dispatch('updateWorldstate')}, 60000)
+/*setInterval(() => {
```
Comment (TobiTenno) [inline src/main.js]: this shouldn't be commented

File: src/components/panels/NewsPanel.vue [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+</template>
+<style>
+.carousel-caption {
+  background-color: rgba(248, 247, 216, 0.6) !important;
+  color: black !important;
+  width: 100% !important;
+  left: 0 !important;
+  bottom: 0 !important;
+  font-weight: bolder !important;
+  padding-bottom: 0 !important;
```
Comment (dsinn) [inline src/components/panels/NewsPanel.vue]: Why is everything `!important`?

File: src/views/Timer.vue [hunk 12]
```diff
@@ -1,18 +1,62 @@
 <template>
   <div class="timers">
     <b-container fluid>
-      <AlertPanel v-if="this.$store.getters.componentState.alerts.state"/>
+          <b-row>
```
Comment (TobiTenno) [inline src/views/Timer.vue]: looks like mixed spaces/tabs or just too much indentation

**Review by TobiTenno — CHANGES_REQUESTED** [review]

**Comment by TobiTenno** [comment]
looks like i missed installing the vue-cli-service on start, so i'll go fix that.

File: src/components/panels/TimePanel.vue [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+      <b-list-group-item :style="styleObject"  class="list-group-item-borderless">
+        <span class="pull-left">
+         <b>Currently it is:</b>
+        </span>
+          <b-badge 
+              class="pull-right"
+              style="margin-right: 5px"
+              variant="dark"><span v-if="time.isDay">Day</span><span v-else>Night</span></b-badge>
+      </b-list-group-item>
+            <b-list-group-item :style="styleObject"  class="list-group-item-borderless">
```
Comment (dsinn) [inline src/components/panels/TimePanel.vue]: The indentation is quite inconsistent in this file.

File: src/components/panels/TimePanel.vue [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+            <h3 class="text-center">{{headertext}}</h3>
+    <b-list-group style="width: 100%">
+      <b-list-group-item :style="styleObject"  class="list-group-item-borderless">
+        <span class="pull-left">
+         <b>Currently it is:</b>
+        </span>
+          <b-badge 
+              class="pull-right"
+              style="margin-right: 5px"
+              variant="dark"><span v-if="time.isDay">Day</span><span v-else>Night</span></b-badge>
```
Comment (dsinn) [inline src/components/panels/TimePanel.vue]: Can the `b-badge` tags go on their own lines?

File: src/components/panels/TimePanel.vue [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+  <b-col md="6">
+            <h3 class="text-center">{{headertext}}</h3>
+    <b-list-group style="width: 100%">
+      <b-list-group-item :style="styleObject"  class="list-group-item-borderless">
+        <span class="pull-left">
+         <b>Currently it is:</b>
+        </span>
+          <b-badge 
+              class="pull-right"
+              style="margin-right: 5px"
```
Comment (dsinn) [inline src/components/panels/TimePanel.vue]: Can we avoid inline styles completely?  I know the old way uses them, but in an overhaul I'd like to get rid of them.

Comment (TobiTenno) [reply src/components/panels/TimePanel.vue]: yeah, that's probably more for me than new on this... which is why i wanted just to review his additions so the rest of the code can get reviewed as we're going on the base branch

File: src/components/panels/TimePanel.vue (hunk 9 above)
Comment (dsinn) [reply src/components/panels/TimePanel.vue]: When I click-drag with the mouse, they all look like spaces.  The highlighting with tab characters should jump.

File: src/views/Timer.vue [hunk 16]
```diff
@@ -1,18 +1,62 @@
...
+  data() {
+    return {
+      platform: this.$store.getters.platform
+    }
+  },
+  mounted() {
+    this.interval = setInterval(
+      function() {
+        this.loadData()
+        console.log('tick')
```
Comment (dsinn) [inline src/views/Timer.vue]: We're going to want to remember to delete this later, if not now.

File: src/components/panels/NewsPanel.vue (hunk 11 above)
Comment (TobiTenno) [reply src/components/panels/NewsPanel.vue]: @Bitti09

File: package.json [hunk 17]
```diff
@@ -13,6 +13,8 @@
     "bootstrap-vue": "^2.0.0-rc.11",
     "bootswatch": "^4.1.3",
     "moment": "^2.22.2",
+    "moment-countdown": "0.0.3",
```
Comment (TobiTenno) [inline package.json]: we already have a "TimeBadge" component

File: src/components/panels/ResetPanel.vue [hunk 18]
```diff
@@ -0,0 +1,61 @@
+<template>
+<b-col md="6">
+  <h3 class="text-center">{{headertext}}</h3>
+  <b-list-group style="width: 100%">
```
Comment (TobiTenno) [inline src/components/panels/ResetPanel.vue]: as @dsinn mentioned, please avoid inline styles

File: src/components/panels/AlertPanel.vue [hunk 19]
```diff
@@ -1,62 +1,51 @@
 <template>
```
Comment (TobiTenno) [inline src/components/panels/AlertPanel.vue]: you haven't reverted the style changes

File: src/components/panels/AlertPanel.vue [hunk 20]
```diff
@@ -1,62 +1,51 @@
...
-  <b-col md="6">
-    <h3 class="text-center">Alerts</h3>
-    <b-list-group>
-      <b-list-group-item :style="styleObject" v-for="alert in getAlerts" :key="alert.id" class="list-group-item-borderless">
```
Comment (TobiTenno) [inline src/components/panels/AlertPanel.vue]: why did you change it from using data to using a prop?

**Review by TobiTenno — CHANGES_REQUESTED** [review]

File: src/components/panels/AlertPanel.vue (hunk 20 above)
Comment (Bitti09) [reply src/components/panels/AlertPanel.vue]: its the way vue does

Comment (TobiTenno) [reply src/components/panels/AlertPanel.vue]: um, no? cause I already had it as data.... _in vue_

File: src/components/panels/NewsPanel.vue [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+  <b-col md="6">
+        <h3 class="text-center">{{headertext}}</h3>
+        <b-carousel id="infoscreen-carousel" :interval="9000" :indicators="false" :controls="true">
+          <b-carousel-slide   v-for="(newsitem,index) in news" :key="index"
+          v-if="newsitem.translations['en']"
+          :text="newsitem.translations['en']">
+          <img slot="img" class="d-block   w-100" width="1024" height="250"
+          :src="newsitem.imageLink" alt="image slot">
+          <p style="color: 'black'!important">
+            <b>Posted: {{newsitem.eta}} | More Info: <a :href="newsitem.link" target="_blank">Click</a>
```
Comment (TobiTenno) [inline src/components/panels/NewsPanel.vue]: need to use a fallback for the newsitem link.
image

File: src/components/panels/TimePanel.vue [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+      <b-badge variant="info" :style="styleObject" class="pull-right" style="margin-right: 5px">{{earthtimezonetime}}</b-badge>
+    </b-list-group-item>
+    <b-list-group-item class="list-group-item-borderbottom">
+    </b-list-group-item>
+  </b-list-group>
+</b-col>
+</template>
+
+<script>
+  import moment from 'moment'
```
Comment (TobiTenno) [inline src/components/panels/TimePanel.vue:35]: use the time badge instead of importing moment again.

File: src/components/panels/ResetPanel.vue [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+  props: ['time', 'location'],
+  name: 'TimePanel',
+  computed: {
+    headertext() {
+      return 'Daily Reward Timer reset'
+    }
+  },
+  data() {
+    return {
+      time1: moment().toISOString(),
```
Comment (TobiTenno) [inline src/components/panels/ResetPanel.vue:29]: use the time badges, not another moment calculation

File: src/views/Timer.vue [hunk 24]
```diff
@@ -1,18 +1,30 @@
 <template>
   <div class="timers">
     <b-container fluid>
-      <AlertPanel v-if="this.$store.getters.componentState.alerts.state"/>
+      <b-row>
+        <AlertPanel v-if="this.$store.getters.componentState.alerts.state" :alerts="this.$store.getters.worldstate.alerts"/>
+        <TimePanel :time="this.$store.getters.worldstate.earthCycle"  :location="'Earth'" v-if="this.$store.getters.componentState.earth.state"/>
```
Comment (TobiTenno) [inline src/views/Timer.vue]: `v-if` first so it's the same order on all the panels so it's easier to tell they're all using the same setup

**Review by TobiTenno — CHANGES_REQUESTED** [review]

File: src/components/panels/NewsPanel.vue [hunk 25]
```diff
@@ -0,0 +1,49 @@
+<template>
+  <b-col md="6">
+        <h3 class="text-center">{{headertext}}</h3>
+        <b-carousel id="infoscreen-carousel" :interval="9000" :indicators="false" :controls="true">
+          <b-carousel-slide   v-for="(newsitem,index) in news" :key="index"
+          v-if="newsitem.translations['en']"
+          :text="newsitem.translations['en']">
+          <img slot="img" class="d-block   w-100" width="1024" height="250"
+          :src="newsitem.imageLink" alt="image slot">
+          <p style="color: 'black'!important">
```
Comment (TobiTenno) [inline src/components/panels/NewsPanel.vue]: please do a dark shade with light text to follow the existing pattern for text on the site.

**Review by TobiTenno — CHANGES_REQUESTED** [review]

File: src/components/panels/NewsPanel.vue (hunk 25 above)
Comment (TobiTenno) [reply src/components/panels/NewsPanel.vue]: alternatively, give both a class so we can change them depending on theme.

File: src/components/panels/ResetPanel.vue (hunk 23 above)
Comment (Bitti09) [reply src/components/panels/ResetPanel.vue:29]: time badage need a start & end date

File: src/components/panels/AlertPanel.vue (hunk 20 above)
Comment (Bitti09) [reply src/components/panels/AlertPanel.vue]: why  do a  round over computed prop  instead of     sending it directly ?

File: src/components/panels/NewsPanel.vue (hunk 21 above)
Comment (Bitti09) [reply src/components/panels/NewsPanel.vue]: sugest one,  bootstrap doesn't  allow  text  only  slides in caroussel

File: src/components/panels/ResetPanel.vue (hunk 23 above)
Comment (Bitti09) [reply src/components/panels/ResetPanel.vue]: and  how tell your badage the end date ?

File: src/components/panels/TimePanel.vue (hunk 22 above)
Comment (Bitti09) [reply src/components/panels/TimePanel.vue:35]: i get a wired -22d   string when using  your  badage  for  the date

File: src/components/panels/ResetPanel.vue (hunk 23 above)
Comment (TobiTenno) [reply src/components/panels/ResetPanel.vue:29]: And why is moment needed for that? that's part of base javascript dates

File: src/components/panels/AlertPanel.vue (hunk 20 above)
Comment (TobiTenno) [reply src/components/panels/AlertPanel.vue]: "a round over"?

the previous way, it's fetched from the state by the component. your way, it's passed to the component, with no validation that it's the right object.

**Review by TobiTenno — CHANGES_REQUESTED** [review]

File: src/components/panels/AlertPanel.vue (hunk 20 above)
Comment (TobiTenno) [reply src/components/panels/AlertPanel.vue]: my point is why did you change the existing pattern without discussion?

File: src/components/panels/NewsPanel.vue (hunk 21 above)
Comment (TobiTenno) [reply src/components/panels/NewsPanel.vue]: https://i.imgur.com/CNrsc7V.png

File: src/components/panels/TimePanel.vue (hunk 22 above)
Comment (TobiTenno) [reply src/components/panels/TimePanel.vue:35]: it's probably sending milliseconds instead of seconds.

Comment (Bitti09) [reply src/components/panels/TimePanel.vue:35]: its getting   the data from the api: expiry:"2018-10-25T00:08:00.000Z"

File: src/components/panels/ResetPanel.vue [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+<script>
+import TimeBadge from '@/components/TimeBadge.vue'
+import moment from 'moment'
+
+export default {
+  props: ['time', 'location'],
+  name: 'TimePanel',
+  computed: {
+    headertext() {
+      return 'Daily Reward Timer reset'
```
[the remark this answers is not in the document]
Comment (TobiTenno) [reply src/components/panels/ResetPanel.vue:24]: I've got this one and others like it automatically fixed on lint. I saw you had git hooks for post-commit or pre-commit or some such.... do you think it'd be valuable to have it run the linter automatically on pushes, @Bitti09?
